cesium/internal/domain/writer.go

```diff
@@ -140,16 +147,35 @@ func (w *Writer) Write(p []byte) (n int, err error) {
 // commit, Commit will return an error. If the domain formed by the WriterConfig.Start
 // and the provided timestamp overlaps with any other domains within the DB, Commit will
 // return an error.
+// If WriterCommit.AutoPersistInterval is greater than 0, then the changes committed would only
+// be persisted to disk after the set interval.
 func (w *Writer) Commit(ctx context.Context, end telem.TimeStamp) error {
+	var (
+		now     = telem.Now()
+		persist = w.AutoPersistInterval <= 0 || w.lastPersist.Span(now) >= w.WriterConfig.AutoPersistInterval
+	)
+
+	if w.AutoPersistInterval > 0 && persist {
```

Comment:
You just checked if the interval is less than or equal to 0. `w.AutoPersistInterval <= 0` will always return false if `w.AutoPersistInterval > 0`, which means your or check is redundant no?